confluent_server/confluent/webauthn.py

```diff
@@ -1,109 +1,399 @@
+    
+    return {"verified": True}
+    
+    
 
 def handle_api_request(url, env, start_response, username, cfm, headers, reqbody, authorized):
+    """
+        For now webauth is going to be limited to just one passkey per user 
+        If you try to register a new passkey this will just clear the old one and regist the new passkey
+    """
+    global CONFIG_MANAGER
+    CONFIG_MANAGER = cfm
     if env['REQUEST_METHOD'] != 'POST':
         raise Exception('Only POST supported for webauthn operations')
     url = url.replace('/sessions/current/webauthn', '')
     if url == '/registration_options':
-        rp = pywarp.RelyingPartyManager('Confluent Web UI', credential_storage_backend=ConfluentBackend(cfm), require_attestation=False)
         userinfo = cfm.get_user(username)
         if not userinfo:
             cfm.create_user(username, role='Stub')
             userinfo = cfm.get_user(username)
-        authid = userinfo.get('authid', None)
+        authid = userinfo.get('webauthid', None)
         if not authid:
-            authid = util.randomstring(64)
-            cfm.set_user(username, {'authid': authid})
-        opts = rp.get_registration_options(username)
-        # pywarp generates an id derived
-        # from username, which is a 'must not' in the spec
-        # we replace that with a complying approach
-        opts['user']['id'] = authid
-        if 'icon' in opts['user']:
-            del opts['user']['icon']
-        if 'id' in opts['rp']:
-            del opts['rp']['id']
+            authid = secrets.token_bytes(64)
+            cfm.set_user(username, {'webauthid': authid})
+        opts = registration_request(username, cfm)
         start_response('200 OK', headers)
         yield json.dumps(opts)
     elif url.startswith('/registered_credentials/'):
         username = url.rsplit('/', 1)[-1]
-        rp = pywarp.RelyingPartyManager('Confluent Web UI', credential_storage_backend=ConfluentBackend(cfm))
+        userinfo = cfm.get_user(username)
         if not isinstance(username, bytes):
             username = username.encode('utf8')
-        opts = rp.get_authentication_options(username)
-        opts['challenge'] = base64.b64encode(opts['challenge']).decode('utf8')
+        opts = authentication_request(username)
         start_response('200 OK', headers)
         yield json.dumps(opts)
     elif url.startswith('/validate/'):
         username = url.rsplit('/', 1)[-1]
+        userinfo = cfm.get_user(username)
         if not isinstance(username, bytes):
             username = username.encode('utf8')
-        rp = pywarp.RelyingPartyManager('Confluent Web UI', credential_storage_backend=ConfluentBackend(cfm))
         req = json.loads(reqbody)
-        for x in req:
-            req[x] = base64.b64decode(req[x].replace('-', '+').replace('_', '/'))
-        req['email'] = username
-        rsp = rp.verify(**req)
+        rsp = authentication_response(req, username)
```

Comment:
Since this authentication response has a non-exceptional return, would make sense to check it for the 'good' path explicitly.  For example, looks like authentication_response could return "Timeout", which validate would then treat as success.